src/DssSpell.t.sol

```diff
@@ -614,369 +604,426 @@ contract DssSpellTest is DssSpellTestBase {
+    GemJoinAbstract          rwa015AJoin            = GemJoinAbstract(addr.addr("MCD_JOIN_RWA015_A"));
+    RwaUrnLike               rwa015AUrn             = RwaUrnLike(addr.addr("RWA015_A_URN"));
+    RwaJarLike               rwa015AJar             = RwaJarLike(addr.addr("RWA015_A_JAR"));
+    RwaOutputConduitLike     rwa015AOutputConduit   = RwaOutputConduitLike(addr.addr("RWA015_A_OUTPUT_CONDUIT"));
+    GemAbstract              psmGem                 = GemAbstract(rwa015AOutputConduit.gem());
+    RwaInputConduitLike      rwa015AInputConduitUrn = RwaInputConduitLike(addr.addr("RWA015_A_INPUT_CONDUIT_URN"));
+    RwaInputConduitLike      rwa015AInputConduitJar = RwaInputConduitLike(addr.addr("RWA015_A_INPUT_CONDUIT_JAR"));
+    GemAbstract              rwa012AGem             = GemAbstract(addr.addr("RWA012"));
+    GemJoinAbstract          rwa012AJoin            = GemJoinAbstract(addr.addr("MCD_JOIN_RWA012_A"));
+
+    uint256 daiPsmGemDiffDecimals               = 10 ** (dai.decimals() - psmGem.decimals());
+
+    // Note: This is an exception because of exceeding the `action` size in the spell. Main pattern is to have this checks in the spell itself
+    function testRWA015_CONTRACT_DEPLOYMENT_SETUP() public {
+        assertEq(rwa015AJoin.vat(), addr.addr("MCD_VAT"),  "join-vat-not-match");
+        assertEq(rwa015AJoin.ilk(), "RWA015-A",            "join-ilk-not-match");
+        assertEq(rwa015AJoin.gem(), address(rwa015AGem),   "join-gem-not-match");
+        assertEq(rwa015AJoin.dec(), rwa015AGem.decimals(), "join-dec-not-match");
+
+        assertEq(rwa015AUrn.vat(),           addr.addr("MCD_VAT"),          "urn-vat-not-match");
+        assertEq(rwa015AUrn.jug(),           addr.addr("MCD_JUG"),          "urn-jug-not-match");
+        assertEq(rwa015AUrn.daiJoin(),       addr.addr("MCD_JOIN_DAI"),     "urn-daijoin-not-match");
+        assertEq(rwa015AUrn.gemJoin(),       address(rwa015AJoin),          "urn-gemjoin-not-match");
+        assertEq(rwa015AUrn.outputConduit(), address(rwa015AOutputConduit), "urn-outputconduit-not-match");
+
+        assertEq(rwa015AJar.chainlog(), addr.addr("CHANGELOG"),    "jar-chainlog-not-match");
+        assertEq(rwa015AJar.dai(),      addr.addr("MCD_DAI"),      "jar-dai-not-match");
+        assertEq(rwa015AJar.daiJoin(),  addr.addr("MCD_JOIN_DAI"), "jar-daijoin-not-match");
+
+        assertEq(rwa015AOutputConduit.dai(), addr.addr("MCD_DAI"),        "output-conduit-dai-not-match");
+        assertEq(rwa015AOutputConduit.gem(), addr.addr("USDC"),           "output-conduit-gem-not-match");
+        assertEq(rwa015AOutputConduit.psm(), addr.addr("MCD_PSM_USDC_A"), "output-conduit-psm-not-match");
+
+        assertEq(rwa015AInputConduitUrn.psm(), addr.addr("MCD_PSM_USDC_A"), "input-conduit-urn-psm-not-match");
+        assertEq(rwa015AInputConduitUrn.to(),  address(rwa015AUrn),         "input-conduit-urn-to-not-match");
```

Comment:
Could do `addr.addr(RWA015_A_URN)` things for these, right? For Urn and Jar on the lines below too.

Reply:
Just use addr.addr instead of address() casting `SomethingLike` objects IMO. Unless there is a drawback to this?

Reply:
Yes, true.

although it changes nothing effectively (afai can see) but requires a lot of lines changed and verified to migrate from one pattern to another. Is it really worth the time and effort?

Reply:
Resolving for now but treat this as a note that changes need to be made at some point in the future